code/controllers/UserController.php

```diff
@@ -66,27 +67,35 @@ public function sendInvite($data, Form $form)
         }
 
         $invite = UserInvitation::create();
-        $form->saveInto($invite);
-        try {
-            $invite->write();
-        } catch (ValidationException $e) {
+        if ($invite->canCreate(Member::CurrentUser()) == true){
```

Comment:
Perhaps move this check to UserInvitation::validate() method as (and also update the comment):
if(!$this->canCreate()){
$valid->error(//your message here);
}